provider/queue/queue.go

```diff
@@ -135,14 +110,13 @@ func (q *Queue) work() {
 
 			select {
 			case toQueue := <-q.enqueue:
-				nextKey := q.queueKey(q.tail)
+				keyPath := fmt.Sprintf("%d/%s", time.Now().UnixNano(), c.String())
```

Comment:
Nit: let's use base32 (`c.StringOfBase` or `c.Encode`). We try to use case-insensitive keys in the datastore.

Reply:
Understood, I did not realize. I'll make this change now.